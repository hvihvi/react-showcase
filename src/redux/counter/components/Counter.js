import React from "react";
import { connect } from "react-redux";
import { onDecrement, onIncrement } from "../redux/counter";

const Counter = ({ value, onDecrement, onIncrement }) => (
  <p>
    Clicked: {value} times
    <button onClick={onIncrement}>+</button>
    <button onClick={onDecrement}>-</button>
  </p>
);

const mapStateToProps = state => {
  return {
    value: state.counter.count
  };
};

const mapDispatchToProps = dispatch => {
  return {
    onIncrement: () => dispatch(onIncrement),
    onDecrement: () => dispatch(onDecrement)
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Counter);
