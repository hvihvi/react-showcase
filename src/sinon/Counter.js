import React from "react";

export const Counter = ({ value, onDecrement, onIncrement }) => (
  <p>
    Clicked: {value} times
    <button onClick={onIncrement}>+</button>
    <button onClick={onDecrement}>-</button>
  </p>
);
