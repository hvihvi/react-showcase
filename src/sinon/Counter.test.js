import Enzyme, { mount, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Counter } from "./Counter";
import renderer from "react-test-renderer";
import React from "react";

Enzyme.configure({ adapter: new Adapter() });

/**
 * Snapshot test example with react-test-renderer to render
 */
it("renders correctly", () => {
  // given
  const tree = renderer
    .create(<Counter value={2} onIncrement={() => {}} onDecrement={() => {}} />)
    .toJSON();
  // then
  expect(tree).toMatchSnapshot();
});

it("should call onIncrement when clicking + buton", () => {
  // given
  const wrapper = shallow(
    <Counter value={2} onIncrement={() => {}} onDecrement={() => {}} />
  );
  // when

  // then
});
