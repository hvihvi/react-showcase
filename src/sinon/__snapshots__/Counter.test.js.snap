// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<p>
  Clicked: 
  2
   times
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
</p>
`;
